### ISSUE TITLE
Prevent argocd from monitoring cluster-scope resource in in-cluster when installing through namespace-install.yaml

### ISSUE THREAD

**Issue by h2ho** [argoproj/argo-cd#4973]
# Summary

When installing argocd using namespace-install.yaml, argocd is not able list cluster-scope resources in in-cluster.  
Argocd seems to monitor resources in cluster-scope at local server.
[https://github.com/argoproj/argo-cd/blob/303925f4a016d7fee80bbe357341df7d3febe298/util/db/cluster.go#L33](url)

Error Message:
```
replicasets.apps is forbidden: User "system:serviceaccount:fr:argocd-application-controller" cannot list resource "replicasets" in API group "apps" at the cluster scope 
```

Workaround: 
1. Add clusterrole with 'list', 'watch' all resources privilege to service account 'application-controller'
2.  Delete in-cluster in argocd and use argocd cli to add in-cluster with namespace option
[https://github.com/argoproj/argo-cd/issues/2176#issuecomment-664726229](url)

# Motivation

Install an Argocd instatnce in a namespace using namespace.install yaml. 
User can deploy resources into the namespace without setting extra config. 

# Proposal

Allows user to specify which namespace to monitor when installing in-cluster

**Comment by h2ho** [comment]
Simple remove Whitelisted cluster resources  in cluster config to prevent list and watch cluster resources

**Comment by alexmt** [comment]
You are right, by default namespaced installation includes RBAC with minimal set of permissions. It does not allow Argo CD to manage resources in the same namespace where Argo CD is installed. The user can provide more permissions by running `argocd cluster add <context> --namespace argocd`.

I think it is better to let user explicitly provide more permissions, don't you think so?

**Comment by alexmt** [comment]
May be we should make documentation more clear?

**Comment by h2ho** [comment]
Maybe I missed something. When installing ArgoCD with namespace-install.yaml, in-cluster is automatically added to the cluster list. I assumed that I would be able to deploy resource to the namespace I installed when I set cluster to in-cluster.  It would definitely help if we could make documentation more clear.

**Comment by abhishekmukherg** [comment]
Not to resurrect a 2y old issue, but would like to agree with @h2ho. If it helps, a use case I'd like to support is that Argo CD can deploy to (several) other kubernetes clusters, while managing its own configuration (AppProjects, Applications) via the [Declarative](https://argo-cd.readthedocs.io/en/stable/operator-manual/declarative-setup/) configuration. In other words, in an ideal world, the Argo CD instance would have

* Admin on several workload clusters
* Get/write/update for relevant objects in _its own namespace_

I may be misunderstanding, but it seems like in order for Argo CD to have an in-cluster-targeting Application, Argo CD would additionally need permission to (at least) get/read all cluster resources as well?
